### PR TITLE
add textual

### PR DESCRIPTION
**Pull request by carlodri**
https://github.com/willmcgugan/textual

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xhochy** [comment]
Build is failing at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366267&view=logs&j=e35d4f76-8ff2-5536-d795-df91e63eb9f7&t=fa7b4b17-b6ff-5c9c-8cfc-15f888c92310 with `ModuleNotFoundError: No module named 'poetry'`

**Comment by carlodri** [comment]
thanks @xhochy! One question: any particular reason for which I can no longer see the details of the CI in the staged-recipes PR page? I only see the linter check...

**Comment by dopplershift** [comment]
@carlodri There were some problems with status reporting which seem to be fixed now?

**Comment by carlodri** [comment]
Yes now I can see the azure pipelines' status in the PR page

File: recipes/textual/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 80ebb32f67489ab88c1eeabca23dc33beac3cef7e978a0acac4df34fdad047ba
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win or py<37]
```
Comment (dopplershift) [inline recipes/textual/meta.yaml:15]: This looks like it's pure Python, so could potentially be noarch Python?
```suggestion
  noarch: python
```

Also, upstream advertises Windows support, so why not here?

File: recipes/textual/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+requirements:
+  host:
+    - python
+    - poetry-core >=1.0.0
+    - pip
+  run:
+    - python
+    - rich >=10.7.0,<11
+    - typing-extensions >=3.10.0,<4  # [py<38]
```
Comment (dopplershift) [inline recipes/textual/meta.yaml:25]: If making noarch, just change to an unconditional dependency:
```suggestion
    - typing-extensions >=3.10.0,<4
```

**Comment by dopplershift** [comment]
My apologies. While the upstream package metadata says it supports Windows, the readme says
> Textual currently runs on MacOS / Linux only. Windows support is in the pipeline.

I've opened an issue about that confusion: willmcgugan/textual#103.

So I guess this needs to return to non-noarch. 😞

**Comment by carlodri** [comment]
@dopplershift no problem! will do, and thanks for the insight.

**Comment by carlodri** [comment]
@willmcgugan are you interested in adding yourself as a package maintainer? 😊

**Comment by carlodri** [comment]
For me this is ready to be merged.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**